woodwork/logical_types.py

```diff
@@ -149,6 +178,19 @@ class Datetime(LogicalType):
     def __init__(self, datetime_format=None):
         self.datetime_format = datetime_format
 
+    def transform(self, series):
+        new_dtype = self._get_valid_dtype(type(series))
+        if new_dtype != str(series.dtype):
+            if dd and isinstance(series, dd.Series):
+                name = series.name
+                series = dd.to_datetime(series, format=self.datetime_format)
+                series.name = name
+            elif ks and isinstance(series, ks.Series):
+                series = ks.Series(ks.to_datetime(series.to_numpy(), format=self.datetime_format), name=series.name)
+            else:
+                series = pd.to_datetime(series, format=self.datetime_format)
```

Comment:
What's the error here if there's an error while converting to `datetime64[ns]`? Before it was getting caught in the try except, so if we've lost the "Error converting datatype" message here, I think it'd be worth adding it back in.

Reply:
I don't mind adding this back in. Did we have a test to check this error?

Reply:
I couldn't find one, but that was probably because we weren't differentiating between datetime and the other logical types with that error, but now we'll need to add it specially to `Datetime.transform` because it wouldn't get caught in the super() call.

Reply:
I can create a test for this. Do we want it in `test_logical_types`?

Reply:
hmmm that's a good question. It looks like we mostly have tested that error through the DataFrame, so `test_invalid_dtype_casting` is in `test_table_accessor`. I'd say if you're going to go through a DataFrame to test it, it should probably be in `test_table_accessor` as well. But if you want to just test something like `Datetime.transform(pd.Series(..))` then it should happen in `test_logical_types`. 

Maybe we should have a set of tests that just test the various transform cases on a series passed to the logical type method instead of always going through the DataFrame.

Reply:
I like the idea of testing various transform cases. It'd help to highlight the expected behavior of a transform.